### PR TITLE
Change absolute URLs to relative in <amp-form> example

### PR DESCRIPTION
**Pull request by ithinkihaveacat**
The `<amp-form>` example uses absolute URLs, but it seems that relative URLs work just fine. (See https://github.com/ampproject/amp-by-example/issues/576.)

The backend server (i.e. `gulp backend:watch`) needs to be running to test this change. Before merging, it would be good to verify that the forms work as expected on the cache too. (I don't know how to deploy to a staging environment to test this.)

Also, for reference (probably only applies to my setup), to get `gulp backend:watch` running, I needed to:

```
# install the google.goland.org/appengine package
$ go get google.golang.org/appengine
# explicitly set the GOROOT and APPENGINE_DEV_APPSERVER env vars
$ GOROOT=$HOME/local/google-cloud-sdk/platform/google_appengine/goroot APPENGINE_DEV_APPSERVER=$(which dev_appserver.py) gulp backend:watch
```

### CODE REVIEW

**Comment by kul3r4** [comment]
If you want to test the forms without deploying to staging, you can build your own instance, for example I use something like [this](https://chiara-ampbyexample.appspot.com/). This is useful so that you can share AMPByExample URLs .

**Comment by sebastianbenz** [comment]
Thanks for doing this. Would you mind changing the other samples as well:

src/50_Samples_%26_Templates/Comment_Section.html
src/50_Samples_%26_Templates/Housing.html
src/50_Samples_%26_Templates/Hotel.html
src/50_Samples_%26_Templates/Poll.html
src/50_Samples_%26_Templates/Product.html
templates/product/header.html

**Comment by ithinkihaveacat** [comment]
@kul3r4 Something like that would be perfect! How do I do that? I can't see anything in the README.

@sebastianbenz Sure, I'll adjust those too. I'll check to see if `<amp-list>` can also be switched to relative.

**Comment by sebastianbenz** [comment]
@ithinkihaveacat create an app engine project and run `goapp deploy -application  my-project-id -version 1`

**Comment by sebastianbenz** [comment]
@ithinkihaveacat  Any updates on this?

**Comment by sebastianbenz** [comment]
@ithinkihaveacat friendly ping

**Comment by ithinkihaveacat** [comment]
Ah sorry, this got filtered somehow. I'll fix this up tomorrow.

**Comment by ithinkihaveacat** [comment]
Updated, and deployed to https://cdn.ampproject.org/c/s/xbyexample.appspot.com/ (https://xbyexample.appspot.com/). Didn't test exhaustively, but everything seems to work.

File: src/50_Samples_%26_Templates/Housing.html [hunk 1]
```diff
@@ -26,7 +26,7 @@
     <script async custom-element="amp-fit-text" src="https://cdn.ampproject.org/v0/amp-fit-text-0.1.js"></script>
     <script async custom-element="amp-analytics" src="https://cdn.ampproject.org/v0/amp-analytics-0.1.js"></script>
 
-    <link rel="canonical" href="<%host%>/sample_amps/housing/">
+    <link rel="canonical" href="/sample_amps/housing/">
```
Comment (sebastianbenz) [inline src/50_Samples_%26_Templates/Housing.html]: afaik canonical need to be fully qualified

File: src/50_Samples_%26_Templates/Product.html [hunk 2]
```diff
@@ -29,7 +29,7 @@
     <script async custom-element="amp-selector" src="https://cdn.ampproject.org/v0/amp-selector-0.1.js"></script>
     <script async custom-element="amp-analytics" src="https://cdn.ampproject.org/v0/amp-analytics-0.1.js"></script>
 
-    <link rel="canonical" href="<%host%>/samples_templates/product/">
+    <link rel="canonical" href="/samples_templates/product/">
```
Comment (sebastianbenz) [inline src/50_Samples_%26_Templates/Product.html]: same as above

File: templates/header.html [hunk 3]
```diff
@@ -6,7 +6,7 @@
   <div id="actions">
     {{#urlSource}}
     {{^metadata.hideCode}}
-    <a href="/playground/#url={{host}}{{urlSource}}" class="button"> 
+    <a href="/playground/#url={{urlSource}}" class="button">
```
Comment (sebastianbenz) [inline templates/header.html]: this breaks the playground

**Comment by ithinkihaveacat** [comment]
Uh, whoops! Thanks, I think we're better now.

**Pull request merged**